support-models/src/main/scala/com/gu/support/workers/states/SendThankYouEmailState.scala

```diff
@@ -2,36 +2,139 @@ package com.gu.support.workers.states
 
 import java.util.UUID
 
-import com.gu.support.encoding.CustomCodecs.{decodeLocalTime, encodeLocalTime}
+import com.gu.support.encoding.Codec
+import com.gu.support.encoding.Codec.deriveCodec
+import com.gu.support.encoding.CustomCodecs._
 import com.gu.support.promotions.PromoCode
-import com.gu.support.redemptions.RedemptionData
+import com.gu.support.workers.GiftRecipient.{DigitalSubscriptionGiftRecipient, WeeklyGiftRecipient}
+import com.gu.support.workers.states.ProductTypeCreated.DigitalSubscriptionCreated._
+import com.gu.support.workers.states.ProductTypeCreated.{ContributionCreated, GuardianWeeklyCreated, PaperCreated}
 import com.gu.support.workers.{PaymentMethod, SalesforceContactRecord, User, _}
+import io.circe.syntax.EncoderOps
+import io.circe.{Decoder, Encoder}
 import org.joda.time.LocalDate
 
 case class SendThankYouEmailState(
   requestId: UUID,
   user: User,
-  giftPurchase: Option[GiftPurchase],
-  product: ProductType,
   analyticsInfo: AnalyticsInfo,
-  paymentOrRedemptionData: Either[PaymentMethodWithSchedule, RedemptionData],
-  firstDeliveryDate: Option[LocalDate],
-  promoCode: Option[PromoCode],
+  productTypeCreated: ProductTypeCreated,
   salesForceContact: SalesforceContactRecord,
-  accountNumber: String,
-  subscriptionNumber: String,
   acquisitionData: Option[AcquisitionData]
 ) extends SendAcquisitionEventState
 
+case class PurchaseInfo(
+  paymentMethod: PaymentMethod,
+  paymentSchedule: PaymentSchedule,
+  promoCode: Option[PromoCode],
+  accountNumber: String,
+  subscriptionNumber: String,
+)
+object PurchaseInfo {
+  implicit val codec: Codec[PurchaseInfo] = deriveCodec
+}
+
+sealed trait ProductTypeCreated {
+  def product: ProductType
+}
+
+object ProductTypeCreated {
+
+  case class ContributionCreated(
+    product: Contribution,
+    purchaseInfo: PurchaseInfo,
+  ) extends ProductTypeCreated
+
+  sealed trait DigitalSubscriptionCreated extends ProductTypeCreated {
+    override def product: DigitalPack
+  }
+
+  sealed trait PurchaseCreated {
+    def purchaseInfo: PurchaseInfo
+  }
+
+  object DigitalSubscriptionCreated {
+
+    case class DigitalSubscriptionDirectPurchaseCreated(
+      product: DigitalPack,
+      purchaseInfo: PurchaseInfo,
+    ) extends DigitalSubscriptionCreated with PurchaseCreated
+
+    case class DigitalSubscriptionGiftPurchaseCreated(
+      product: DigitalPack,
+      giftRecipient: DigitalSubscriptionGiftRecipient,
+      giftCode: GeneratedGiftCode,
+      lastRedemptionDate: LocalDate,
+      purchaseInfo: PurchaseInfo,
+    ) extends DigitalSubscriptionCreated with PurchaseCreated
+
+    case class DigitalSubscriptionCorporateRedemptionCreated(
+      product: DigitalPack,
+      subscriptionNumber: String,
+    ) extends DigitalSubscriptionCreated
+
+    case class DigitalSubscriptionGiftRedemptionCreated( //tbc
```

Comment:
this is where the new fields will be added for start/end date of the gift.  (Start date would be today, and end date would be either 3 or 12 months forward)